src/AppInstallerRepositoryCore/Microsoft/PortableIndex.cpp

```diff
@@ -26,6 +26,33 @@ namespace AppInstaller::Repository::Microsoft
         return result;
     }
 
+    Schema::IPortableIndex::PortableFile PortableIndex::CreatePortableFileFromPath(const std::filesystem::path& path)
+    {
+        Schema::IPortableIndex::PortableFile portableFile;
+        portableFile.SetFilePath(path);
+
+        if (std::filesystem::is_directory(path))
+        {
+            portableFile.FileType = Schema::IPortableIndex::PortableFileType::Directory;
+        }
+        else if (std::filesystem::is_symlink(path))
+        {
+            portableFile.FileType = Schema::IPortableIndex::PortableFileType::Symlink;
+            portableFile.SymlinkTarget = std::filesystem::read_symlink(path).u8string();
+        }
+        else
+        {
+            portableFile.FileType = Schema::IPortableIndex::PortableFileType::File;
+
+            std::ifstream inStream{ path, std::ifstream::binary };

```

Comment:
This seems to be the same code for getti g the File hash as the VerifyPortableExe. Perhaps there should be a SHA256::GetFileHash method added to Utility that returns the hash given a path